pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -601,7 +615,7 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(pInfo *framework.QueuedPodI
 	if inFlightMoveRequestCycle > moveRequestCycle {
 		moveRequestCycle = inFlightMoveRequestCycle
 	}
-	if moveRequestCycle >= podSchedulingCycle {
+	if moveRequestCycle >= podSchedulingCycle && p.inFlightPods[pInfo.Pod.UID].successRequeuePlugins.HasAny(pInfo.UnschedulablePlugins.UnsortedList()...) {
```

Comment:
Can you explain why this is correct?

The rationale behind this check is not immediately obvious to me.

Reply:
`inFlightPod.successRequeuePlugins` has plugins that may change its scheduling result by events that happened during scheduling. 

If `pInfo.UnschedulablePlugins` has more than one of `inFlightPod.successRequeuePlugins`, that means that "the Pod is failed by pluginA, but pluginA may change scheduling result if try again now.".